### DISCUSSION TITLE
Could not push metrics to prometheus gateway

### DISCUSSION THREAD

**Discussion by ryanchen34057** [seaweedfs/seaweedfs#1812] [Q&A]
I was trying to configure system metrics using docker compose file in [here](https://github.com/chrislusf/seaweedfs/tree/master/docker).

I used this [repo](https://github.com/evnsio/prom-stack) to set up prometheus gateway and grafana.

but in the console log:
`volume_1      | I0218 02:18:00     1 metrics.go:162] could not push metrics to prometheus push gateway localhost:9091: Put "http://localhost:9091/metrics/job/volumeServer/instance/172.20.0.3%3A8080": dial tcp 127.0.0.1:9091: connect: connection refused
filer_1       | I0218 02:18:00     1 metrics.go:162] could not push metrics to prometheus push gateway localhost:9091: Put "http://localhost:9091/metrics/job/filer/instance/9f000c390bd6%3A8888": dial tcp 127.0.0.1:9091: connect: connection refused
s3_1          | I0218 02:18:01     1 metrics.go:162] could not push metrics to prometheus push gateway localhost:9091: Put "http://localhost:9091/metrics/job/s3/instance/658aa8024458%3A8333": dial tcp 127.0.0.1:9091: connect: connection refused
volume_1      | I0218 02:18:15     1 metrics.go:162] could not push metrics to prometheus push gateway localhost:9091: Put "http://localhost:9091/metrics/job/volumeServer/instance/172.20.0.3%3A8080": dial tcp 127.0.0.1:9091: connect: connection refused
filer_1       | I0218 02:18:15     1 metrics.go:162] could not push metrics to prometheus push gateway localhost:9091: Put "http://localhost:9091/metrics/job/filer/instance/9f000c390bd6%3A8888": dial tcp 127.0.0.1:9091: connect: connection refused
s3_1          | I0218 02:18:16     1 metrics.go:162] could not push metrics to prometheus push gateway localhost:9091: Put "http://localhost:9091/metrics/job/s3/instance/658aa8024458%3A8333": dial tcp 127.0.0.1:9091: connect: connection refused
volume_1      | I0218 02:18:30     1 metrics.go:162] could not push metrics to prometheus push gateway localhost:9091: Put "http://localhost:9091/metrics/job/volumeServer/instance/172.20.0.3%3A8080": dial tcp 127.0.0.1:9091: connect: connection refused
filer_1       | I0218 02:18:30     1 metrics.go:162] could not push metrics to prometheus push gateway localhost:9091: Put "http://localhost:9091/metrics/job/filer/instance/9f000c390bd6%3A8888": dial tcp 127.0.0.1:9091: connect: connection refused
s3_1          | I0218 02:18:31     1 metrics.go:162] could not push metrics to prometheus push gateway localhost:9091: Put "http://localhost:9091/metrics/job/s3/instance/658aa8024458%3A8333": dial tcp 127.0.0.1:9091: connect: connection refused
volume_1      | I0218 02:18:45     1 metrics.go:162] could not push metrics to prometheus push gateway localhost:9091: Put "http://localhost:9091/metrics/job/volumeServer/instance/172.20.0.3%3A8080": dial tcp 127.0.0.1:9091: connect: connection refused
filer_1       | I0218 02:18:45     1 metrics.go:162] could not push metrics to prometheus push gateway localhost:9091: Put "http://localhost:9091/metrics/job/filer/instance/9f000c390bd6%3A8888": dial tcp 127.0.0.1:9091: connect: connection refused
s3_1          | I0218 02:18:46     1 metrics.go:162] could not push metrics to prometheus push gateway localhost:9091: Put "http://localhost:9091/metrics/job/s3/instance/658aa8024458%3A8333": dial tcp 127.0.0.1:9091: connect: connection refused
volume_1      | I0218 02:19:00     1 metrics.go:162] could not push metrics to prometheus push gateway localhost:9091: Put "http://localhost:9091/metrics/job/volumeServer/instance/172.20.0.3%3A8080": dial tcp 127.0.0.1:9091: connect: connection refused
filer_1       | I0218 02:19:00     1 metrics.go:162] could not push metrics to prometheus push gateway localhost:9091: Put "http://localhost:9091/metrics/job/filer/instance/9f000c390bd6%3A8888": dial tcp 127.0.0.1:9091: connect: connection refused
s3_1          | I0218 02:19:01     1 metrics.go:162] could not push metrics to prometheus push gateway localhost:9091: Put "http://localhost:9091/metrics/job/s3/instance/658aa8024458%3A8333": dial tcp 127.0.0.1:9091: connect: connection refused
volume_1      | I0218 02:19:15     1 metrics.go:162] could not push metrics to prometheus push gateway localhost:9091: Put "http://localhost:9091/metrics/job/volumeServer/instance/172.20.0.3%3A8080": dial tcp 127.0.0.1:9091: connect: connection refused
filer_1       | I0218 02:19:15     1 metrics.go:162] could not push metrics to prometheus push gateway localhost:9091: Put "http://localhost:9091/metrics/job/filer/instance/9f000c390bd6%3A8888": dial tcp 127.0.0.1:9091: connect: connection refused
s3_1          | I0218 02:19:16     1 metrics.go:162] could not push metrics to prometheus push gateway localhost:9091: Put "http://localhost:9091/metrics/job/s3/instance/658aa8024458%3A8333": dial tcp 127.0.0.1:9091: connect: connection refused
`
Does anyone know what I did wrong here?

**Comment by chrislusf** [comment]
you need to show you what you did first. Just description is not enough.
